### PR TITLE
Add optimization page

### PR DESCRIPTION
**Pull request by angrykoala**
Adds an optimisation page to cover steps that users may do to improve their system's performance

It covers the optimization added: https://github.com/neo4j/graphql/pull/5104

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-graphql-145.surge.sh

New pages:
- [Optimization](https://neo4j-docs-graphql-145.surge.sh/graphql/5/optimization)

File: modules/ROOT/pages/optimization.adoc [hunk 1]
```diff
@@ -0,0 +1,35 @@
+[[optimization]]
+= Optimization
+
+
+This chapter contains some optimizations to improve the performance of a production system using `@neo4j/graphql`.
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/optimization.adoc]:
```suggestion
This page contains optimizations to improve the performance of a production system using `@neo4j/graphql`.
```

File: modules/ROOT/pages/optimization.adoc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+== Schema Optimisations
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/optimization.adoc]:
```suggestion
== Schema optimizations
```

File: modules/ROOT/pages/optimization.adoc [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+This section covers improvements to reduce the size of the generated schema. Reducing the schema size has the following performance benefits:
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/optimization.adoc]:
```suggestion
This section covers optimizations to reduce the size of the generated schema. Reducing the schema size has the following performance benefits:
```

File: modules/ROOT/pages/optimization.adoc [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+* Reduce the startup time of the server.
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/optimization.adoc]:
```suggestion
* Reduce server startup time.
```

File: modules/ROOT/pages/optimization.adoc [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+* Reduce memory footprint.
+
+=== Exclude `@deprecated` fields
+The `@neo4j/graphql` library generates some GraphQL fields and operations marked as `@deprecated`. These exists to keep compatibility with previous versions of the library.
+
+If you are not using these deprecated fields, you can disable generation for these with the `excludeDeprecatedFields` flag in the library setup.
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/optimization.adoc]:
```suggestion
If you are not using these deprecated fields, you can disable their generation with the `excludeDeprecatedFields` flag in the library setup. 
```

File: modules/ROOT/pages/optimization.adoc (hunk 2 above)
Comment (rsill-neo4j) [inline modules/ROOT/pages/optimization.adoc]: Will there be more optimization sections on this page in the future?

Comment (angrykoala) [reply modules/ROOT/pages/optimization.adoc]: Yes, the plan is to add a few things that we are planning, for "Schema" and "Queries"

**Comment by neo-technology-commit-status-publisher** [comment]
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.

**Review by rsill-neo4j — APPROVED** [review]

**Pull request merged**